String.php

```diff
@@ -755,32 +790,59 @@ public function getWidth ( ) {
     public function getDirection ( ) {
 
         if(null === $this->_direction) {
-
-            if(null === $this->_string)
-                $this->_direction = static::LTR;
-            else
-                $this->_direction = static::getCharDirection(
-                    mb_substr($this->_string, 0, 1)
-                );
+            // Default
+            $this->_direction = static::LTR;
+
+            // Check for LRM or RLM/ARM
+            $hasLRM      = 0 !== preg_match('#\x{200e}#u', $this->_string);
```

Comment:
The problem here is, that the already defined static constant (LRM, RLM, ARM, etc) cannot be easily included in the regular expression, as they are only integers ans regular expression use a `\x{codepoint` syntax - given the `u` modifier at the end. So no easy solution for that..